packages/combo-box/src/vaadin-combo-box-mixin.js

```diff
@@ -999,12 +997,13 @@ export const ComboBoxMixin = (subclass) =>
       const valueIndex = this._indexOfValue(value, this.filteredItems);
       const previouslySelectedItem = this.selectedItem;
 
-      this.selectedItem =
-        valueIndex >= 0
-          ? this.filteredItems[valueIndex]
-          : this.dataProvider && this.selectedItem === undefined
-          ? undefined
-          : null;
+      if (valueIndex >= 0) {
```

Comment:
**note:** It was difficult to read these conditions before so I turned them into the regular `if` / `else`.